src/cmd/services/m3coordinator/ingest/m3msg/ingest.go

```diff
@@ -223,15 +230,53 @@ func (op *ingestOp) resetWriteQuery() error {
 			Resolution:  op.sp.Resolution().Window,
 			Retention:   op.sp.Retention().Duration(),
 		},
+	}
+
+	if op.storeMetricsType {
+		var err error
+		wq.Annotation, err = op.convertTypeToAnnotation(op.metricType)
+		if err != nil {
+			return err
+		}
+	}
+
+	return op.writeQuery.Reset(wq)
+}
+
+func (op *ingestOp) convertTypeToAnnotation(tp ts.PromMetricType) ([]byte, error) {
+	if tp == ts.PromMetricTypeUnknown {
+		return nil, nil
+	}
+
+	handleValueResets := false
+	if tp == ts.PromMetricTypeCounter {
+		handleValueResets = true
+	}
+
+	annotationPayload, err := storage.SeriesAttributesToAnnotationPayload(ts.SeriesAttributes{
+		PromType:          tp,
+		HandleValueResets: handleValueResets,
```

Comment:
It would be cleaner if `SeriesAttributesToAnnotationPayload` now took those two individual arguments and not the `ts.SeriesAttributes` (which probably does not fit well in this part of code). Then if some third field is added, the compiler will check that we need to pass it here as well. With struct, we lose that safety.